i6_models/config.py

```diff
@@ -0,0 +1,56 @@
+"""
+Provides the base class for configurations, model configurations can be derived from this base class e.g:
+
+@dataclass
+class ExampleConfig(ModelConfiguration):
+    hidden_dim: int = 256
+    name: str = "Example Configuration"
+
+class ExampleModule(Module):
+    __init__(self, cfg: ExampleConfig)
+        self.hidden_dim = cfg.hidden_dim
+
+This config can then be used in the construction of the model to provide parameters.
+Approach is inspired by the Fairseq: https://github.com/facebookresearch/fairseq/blob/main/fairseq/dataclass/configs.py
+"""
```

Comment:
Add empty line. And also import future annotation support.
```suggestion
"""

from __future__ import annotations
```